src/Uno.UI.Composition/Composition/SkiaCompositionSurface.skia.cs

```diff
@@ -1,27 +1,39 @@
 ﻿#nullable enable
 
-
 using SkiaSharp;
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
 using System.Runtime.CompilerServices;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 using Uno.Extensions;
 using Uno.Foundation.Logging;
+using Uno.UI.Dispatching;
 using Windows.Graphics;
 
 namespace Microsoft.UI.Composition
 {
-	internal partial class SkiaCompositionSurface : ICompositionSurface
+	internal partial class SkiaCompositionSurface : CompositionObject, ICompositionSurface
 	{
-		private SKImage? _image;
+		private SKImage[]? _images;
+		private SKCodecFrameInfo[]? _infos;
```

Comment:
Why do we keep the `_infos` in a private field? This could be kept as a capture in the timer callback and here it will leak if we invoke `LoadFromStream` again with a still image (not sure if possible).

You should instead keep the instance of the `Timer` and stop it if we invoke the `LoadFromStream` again or this object is being disposed (note: currently this object will leak until the all frames have been consumed).